server/API/upload_functions/UploadTeachers.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+define("TEACHER_ID", "Number");
+define("TEACHER_NAME", "GivenName");
+define("TEACHER_SURNAME", "Surname");
+define("TEACHER_EMAIL", "OfficialEmail");
+define("TEACHER_SSN", "SSN");
+
+if (!function_exists("upload_teachers")) {
+	function upload_teachers($vars) {
+		try {
+			$logged_user = get_myself();
+
+			if (intval($logged_user['type']) != USER_TYPE_SPRT_OFCR) {
+				throw new ErrorException("Wrong permissions");
+			}
+
+
+			// Undefined | Multiple Files | $_FILES Corruption Attack
+			// If this request falls under any of them, treat it invalid.
+			if (
+				!isset($_FILES['teacher_file']['error']) ||
+				is_array($_FILES['teacher_file']['error'])
+			) {
+				throw new RuntimeException('Invalid parameters.');
+			}
+
+			// Check $_FILES['teacher_file']['error'] value.
+			switch ($_FILES['teacher_file']['error']) {
+				case UPLOAD_ERR_OK:
+					break;
+				case UPLOAD_ERR_NO_FILE:
+					throw new RuntimeException('No file sent.');
+				case UPLOAD_ERR_INI_SIZE:
+				case UPLOAD_ERR_FORM_SIZE:
+					throw new RuntimeException('Exceeded filesize limit.');
+				default:
+					throw new RuntimeException('Unknown errors.');
+			}
+
+			$csv_file = array_map('str_getcsv', str_getcsv(file_get_contents($_FILES['teacher_file']['tmp_name']), "\n"));
+
+			$positions = array(
+				TEACHER_ID => array_search(TEACHER_ID, $csv_file[0]),
+				TEACHER_NAME => array_search(TEACHER_NAME, $csv_file[0]),
+				TEACHER_SURNAME => array_search(TEACHER_SURNAME, $csv_file[0]),
+				TEACHER_EMAIL => array_search(TEACHER_EMAIL, $csv_file[0]),
+				TEACHER_SSN => array_search(TEACHER_SSN, $csv_file[0])
+			);
+
+			if (array_search(false, $positions) === FALSE) {
```

Comment:
This line is wrong. I corrected it in the other files.
```suggestion
			if (array_search(FALSE, $positions, TRUE) !== FALSE) {
```